From f278746cd0c0428ddb12ade854e5efb1ee657c76 Mon Sep 17 00:00:00 2001
From: Dawid Trendota <dawid.trendota@consult.red>
Date: Thu, 27 Apr 2023 14:09:23 +0200
Subject: [PATCH] fix tests compilation

- Update CMAKE_MODULE_PATH for cryptography_test
- revert removal of Helpers Teardown method
---
 Source/cryptography/tests/cryptography_test/CMakeLists.txt | 2 ++
 Source/cryptography/tests/cryptography_test/Helpers.cpp    | 5 +++++
 Source/cryptography/tests/cryptography_test/Helpers.h      | 2 ++
 3 files changed, 9 insertions(+)

diff --git a/Source/cryptography/tests/cryptography_test/CMakeLists.txt b/Source/cryptography/tests/cryptography_test/CMakeLists.txt
index 873382a..9ce2656 100644
--- a/Source/cryptography/tests/cryptography_test/CMakeLists.txt
+++ b/Source/cryptography/tests/cryptography_test/CMakeLists.txt
@@ -15,6 +15,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+set(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/cmake")
+
 option(BUILD_NETFLIX_VAULT_GENERATOR "Build Netflix Vault generator" OFF)
 
 
diff --git a/Source/cryptography/tests/cryptography_test/Helpers.cpp b/Source/cryptography/tests/cryptography_test/Helpers.cpp
index 5369a1d..4c8d69b 100644
--- a/Source/cryptography/tests/cryptography_test/Helpers.cpp
+++ b/Source/cryptography/tests/cryptography_test/Helpers.cpp
@@ -199,4 +199,9 @@ bool DHAuthenticatedDerive(DH* dh VARIABLE_IS_NOT_USED, const uint16_t secretSiz
     return (true);
 }
 
+void Teardown()
+{
+    WPEFramework::Core::Singleton::Dispose();
+}
+
 } // extern "C"
diff --git a/Source/cryptography/tests/cryptography_test/Helpers.h b/Source/cryptography/tests/cryptography_test/Helpers.h
index 18bbfb4..c976b5f 100644
--- a/Source/cryptography/tests/cryptography_test/Helpers.h
+++ b/Source/cryptography/tests/cryptography_test/Helpers.h
@@ -44,6 +44,8 @@ bool DHAuthenticatedDerive(DH* dh, const uint16_t secretSize, const uint8_t secr
                            const uint16_t saltSize, const uint8_t salt[], const uint16_t dataSize, const uint8_t data[],
                            uint8_t* encKeyOut[], uint8_t* hmacKeyOut[], uint8_t* wrapKeyOut[]);
 
+void Teardown();
+
 #ifdef __cplusplus
 }
 #endif
-- 
2.25.1

